packages/components/src/border-control/test/index.js

```diff
@@ -415,53 +407,62 @@ describe( 'BorderControl', () => {
 		} );
 
 		it( 'should clear color and set style to `none` when setting zero width', async () => {
+			const user = userEvent.setup();
 			const props = createProps();
 			render( <BorderControl { ...props } /> );
-			await openPopover();
-			selectColorOption( 'Color: Green' );
-			clickButton( 'Dotted' );
-			setWidthInput( '0' );
+			await openPopover( user );
+			await user.click( getColorOption( 'Green' ) );
+			await user.click( getButton( 'Dotted' ) );
+			const widthInput = getWidthInput();
+			await user.clear( widthInput );
+			await user.type( widthInput, '0' );
 
-			expect( props.onChange ).toHaveBeenNthCalledWith( 3, {
+			expect( props.onChange ).toHaveBeenNthCalledWith( 4, {
```

Comment:
What is causing the `onChange` prop to be called one more time? Could it be related to the fact that [`user.type` also clicks on the element](https://testing-library.com/docs/user-event/utility#type)? If that's the case, we could try passing the `skipClick` option, or use `user.key` instead (since the input already has focus).

Same applies to other instances of this PR where the number of times that the `onChange` prop was called changed.

Reply:
It's because of the additional `user.clear` call, which generates a `change` event of its own. Without it, `user.type` appends to the current value, rather than replacing it (which makes sense).

Did a bit of digging though, and `user.type` also accepts a selection range, which means the value of the field is totally replaced. This is closer to the original intent of directly firing a `change` event, so while `user.clear => user.type` is probably closer to most user interactions, `user.type` with a selection is probably preferable.

```javascript
await user.type( getWidthInput(), '0', {
  initialSelectionStart: 0,
  initialSelectionEnd: 1,
} );
```

Reply:
Thank you for the digging, your proposed solution looks good :)